tk_toolchain/cmd_line_tools/tk_run_app/__init__.py

```diff
@@ -130,6 +135,70 @@ def _start_engine(repo, entity_type, entity_id, config):
     return engine
 
 
+def _is_app_command(info):
+    """
+    Checks if an app is an application command.
+
+    :returns: ``True`` if this is an app command, ``False`` otherwise.
+    """
+    # Application commands always have properties.
+    if "app" not in info["properties"]:
+        # Certain commands are not coming from apps, so skip those for now.
+        return False
+    # Make sure that the application command is from the app in the current repo.
+    return info["properties"]["app"].instance_name == "tk-multi-run-this-app"
+
+
+def _get_available_commands(engine):
+    """
+    Gets all available commands, long and short names, prints them on screen
+    and returns them.
+
+    :param engine: The Toolkit engine for which we desire to extract commands.
+
+    :returns: ``list`` of command ``str`` names.
+    """
+    possible_commands = sorted(
+        name for name, info in engine.commands.items() if _is_app_command(info)
+    )
+
+    # Add the list of short commands, sorted.
+    possible_commands += sorted(
+        info["properties"]["short_name"]
+        for info in engine.commands.values()
+        if _is_app_command(info)
+    )
+
+    print("Available application commands (long and short versions):")
+    pprint(possible_commands)
+
+    return possible_commands
+
+
+def _validate_requested_commands(commands, available_commands, engine):
+    """
+    Gets the list of commands the user wishes to execute, prints them
+    and returns them.
+
+    :param commands: List of commands the user typed in.
+    :param available_commands: List of commands, short and long, available.
+    :param engine: The Toolkit engine for which we desire to validate commands.
+
+    :returns: ``list`` of command ``str`` names.
+    """
+    for command in commands:
+        if command not in available_commands:
+            raise SystemExit("Command '%s' does not exist." % command)
+
+    print("The following commands will be run:")
+    if commands:
+        pprint(commands)
+        return commands
+    else:
+        print("all")
```

Comment:
Would it make sense to print the actual commands here as well? That way if you're not passing a command list but you're trying to debug why a certain command didn't run, you can check this print to make sure it was registered and picked up correctly.